tensorflow-ops/src/TensorFlow/Ops.hs

```diff
@@ -151,37 +194,44 @@ instance ( TensorType a
     signum = CoreOps.sign
     negate = CoreOps.neg
 
-matTranspose :: forall a v . TensorType a
-             => Tensor v a -> Tensor Value a
-matTranspose = flip CoreOps.transpose (vector [1, 0 :: Int32])
+matTranspose :: TensorType a => Tensor v a -> Tensor Value a
+matTranspose = matTranspose' id
+
+matTranspose' :: TensorType a => OpParams -> Tensor v a -> Tensor Value a
+matTranspose' params = flip (CoreOps.transpose' params) (vector [1, 0 :: Int32])
 
-placeholder :: forall a m . (MonadBuild m, TensorType a) => Shape -> m (Tensor Value a)
-placeholder shape' =
-    build $ buildOp $ opDef "Placeholder"
-            & opAttr "dtype" .~ tensorType (undefined :: a)
-            & opAttr "shape" .~ shape'
+placeholder :: (MonadBuild m, TensorType a) => Shape -> m (Tensor Value a)
+placeholder = placeholder' id
+
+placeholder' :: (MonadBuild m, TensorType a) => OpParams -> Shape -> m (Tensor Value a)
+placeholder' params pShape
+    = render $ CoreOps.placeholder' (params . (opAttr "shape" .~ pShape))
```

Comment:
The new version of placeholder can get CSE'd. Not sure what a good solution is besides reverting or having list of opgen overrides.